### PR TITLE
[KYUUBI #1936][FOLLOWUP] Send credentials when opening session and wait for completion

### PR DESCRIPTION
**Pull request by lightning-L**
### _Why are the changes needed?_

Follow up #1936 

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by lightning-L** [comment]
I am working on resolving the test failures.

**Comment by yaooqinn** [comment]
@lightning-L thanks, just take your time.

File: kyuubi-server/src/main/scala/org/apache/kyuubi/credentials/HadoopCredentialsManager.scala [hunk 1]
```diff
@@ -212,17 +215,18 @@ class HadoopCredentialsManager private (name: String) extends AbstractService(na
...
-          providers.values
-            .foreach(_.obtainDelegationTokens(userRef.getAppUser, creds))
-          userRef.updateCredentials(creds)
+          val credentialsFuture: Future[Unit] = Future {
+            val creds = new Credentials()
+            providers.values
+              .foreach(_.obtainDelegationTokens(userRef.getAppUser, creds))
+            userRef.updateCredentials(creds)
+          }(ExecutionContext.fromExecutorService(renewalExecutor.get))
+          userRef.setFuture(credentialsFuture)
```
Comment (turboFei) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/credentials/HadoopCredentialsManager.scala:228]: shall we wait the future completion here?

**Review by ulysses-you — APPROVED** [review]

**Comment by ulysses-you** [comment]
thanks, merging to master/branch-1.5

**Pull request closed without merging**